kubernetes/resource_kubernetes_deployment_test.go

```diff
@@ -875,29 +899,67 @@ func testAccCheckKubernetesDeploymentDestroy(s *terraform.State) error {
 	return nil
 }
 
-func testAccCheckKubernetesDeploymentExists(n string, obj *api.Deployment) resource.TestCheckFunc {
-	return func(s *terraform.State) error {
-		rs, ok := s.RootModule().Resources[n]
-		if !ok {
-			return fmt.Errorf("Not found: %s", n)
-		}
+func getDeploymentFromResourceName(s *terraform.State, n string) (*appsv1.Deployment, error) {
+	rs, ok := s.RootModule().Resources[n]
+	if !ok {
+		return nil, fmt.Errorf("Not found: %s", n)
+	}
+
+	conn, err := testAccProvider.Meta().(KubeClientsets).MainClientset()
+	if err != nil {
+		return nil, err
+	}
+
+	namespace, name, err := idParts(rs.Primary.ID)
+	if err != nil {
+		return nil, err
+	}
 
-		conn, err := testAccProvider.Meta().(KubeClientsets).MainClientset()
+	out, err := conn.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func testAccCheckKubernetesDeploymentExists(n string, obj *appsv1.Deployment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		d, err := getDeploymentFromResourceName(s, n)
 		if err != nil {
 			return err
 		}
+		*obj = *d
```

Comment:
Since these are both already pointers, I was surprised to find that the `*` is necessary. But it would seem you're right, judging by the results of the acceptance tests:

```
[dakini@dax kubernetes (deployment-wait *$=)]$ git diff
diff --git a/kubernetes/resource_kubernetes_deployment_test.go b/kubernetes/resource_kubernetes_deployment_test.go
index 6981e8bf..04aa8bc4 100644
--- a/kubernetes/resource_kubernetes_deployment_test.go
+++ b/kubernetes/resource_kubernetes_deployment_test.go
@@ -929,7 +929,7 @@ func testAccCheckKubernetesDeploymentExists(n string, obj *appsv1.Deployment) re
                if err != nil {
                        return err
                }
-               *obj = *d
+               obj = d
                return nil
        }
 }

[dakini@dax kubernetes (deployment-wait *$=)]$ TF_ACC=true go test -v -run TestAccKubernetesDeployment_basic
=== RUN   TestAccKubernetesDeployment_basic
    TestAccKubernetesDeployment_basic: testing.go:654: Step 0 error: Check failed: 2 errors occurred:
                * Check 6/22 error:  annotations don't match.
        Expected: map["TestAnnotationOne":"one" "TestAnnotationTwo":"two"]
        Given: map[]
                * Check 11/22 error:  labels don't match.
        Expected: map["TestLabelOne":"one" "TestLabelThree":"three" "TestLabelTwo":"two"]
        Given: map[]
```

Looks like I have some studying to do.